### PR TITLE
Using the new TreeRecursion extensions for MailExplorer

### PR DESCRIPTION
**Pull request by legionth**

### CODE REVIEW

File: Services/Mail/classes/class.ilMailExplorer.php [hunk 1]
```diff
@@ -108,4 +108,50 @@ function isNodeHighlighted($a_node)
...
+    ) {
+        global $DIC;
+
+        $path = $this->getNodeIcon($node);
+
+        $icon = $DIC->ui()
+            ->factory()
+            ->symbol()
+            ->icon()
+            ->custom($path, 'something');
```
Comment (mjansenDatabay) [inline Services/Mail/classes/class.ilMailExplorer.php]: `something` ? If this argument is used for the `alt` tag of  `img` HTML elements, please use the mail folder title instead.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Only one tiny issue.

In General: For the final implementation, we shoul get rid of the `ilTreeExplorerGUI` inheritance.

**Review by mjansenDatabay — APPROVED** [review]
LGTM, thx: Please remove the "[WIP"] prefix and resolve the reaming conflict.

**Comment by mjansenDatabay** [comment]
@legionth There is one remaining conflict :-).

**Comment by legionth** [comment]
:roll_eyes: I'll fix it.

**Comment by legionth** [comment]
Hm, I would need to rebase `feature-mail-tree-slate` for the current trunk first.
Is this ok for you or should we about another way to release this for the core?

**Comment by legionth** [comment]
Rebased :tada:

**Pull request merged**